x/evm/keeper/fees.go

```diff
@@ -55,8 +57,26 @@ func (k *Keeper) DeductTxCostsFromUserBalance(
 		return errorsmod.Wrapf(err, "account not found for sender %s", from)
 	}
 
+	// Convert the balance to 18 decimals
+	signerBalance := k.bankKeeper.GetBalance(ctx, signerAcc.GetAddress(), utils.BaseDenom)
+	balanceIn18Decimals := utils.ConvertTo18Decimals(*signerBalance.Amount.BigInt())
+
+	found, feeToken := fees.Find(types.DefaultEVMDenom)
```

Comment:
ditto get `evmDenom` from params
```suggestion
	found, feeToken := fees.Find(evmDenom)
```